### PR TITLE
feat: Implemented border when hover

### PR DESCRIPTION
**Pull request by rishabdev2997**
### **feat: Highlight or add left border to active navbar subitem** ### 

#487 - added the feature.

In the issue the problem was not mentioned properly(the download nav item gets a left border for distinction. The same can apply to nav sub-items.) but it seems it indicates navs under the "Table of Contents" section must display the same left border upon selection. 

### **How to see the changes**
- Go to documentation.
- Go the environment.
- Now move your mouse over the contents of the table.

### **Here is the Screenshot**

https://github.com/user-attachments/assets/04af6693-633f-4602-90c8-3399463e9384

### CODE REVIEW

**Comment by rishabdev2997** [comment]
### **Added spacing between each content items in the list**
Updated Screenshot for #508 

https://github.com/user-attachments/assets/9e4218f9-dd8f-4c6f-b2df-c2c9358eebb7

File: package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "react-dom": "^17.0.2",
     "react-helmet": "^6.1.0",
     "react-intl": "^5.25.1",
+    "react-refresh": "^0.14.2",
```
Comment (Stefterv) [inline package.json:74]: Not part of this pr

File: src/components/Sidebar.module.css [hunk 2]
```diff
@@ -110,20 +110,40 @@
 }
 
 .tocItem {
+  position: relative;
+  margin-bottom: 0.5rem;
+
   & a {
     font-family: var(--font);
-    text-overflow: ellipsis;
```
Comment (Stefterv) [inline src/components/Sidebar.module.css:115]: should stay

File: src/components/Sidebar.module.css [hunk 3]
```diff
@@ -110,20 +110,40 @@
...
+    padding: 3px 0.75rem;
+    position: relative;
+    transition: color 0.3s, border-left-color 0.3s;
+
     &:hover {
       color: var(--processing-blue-mid);
+      border-left-color: var(--processing-blue-mid);
+    }
+
+    &::before {
```
Comment (Stefterv) [inline src/components/Sidebar.module.css:131]: This is unused

**Review by Stefterv — CHANGES_REQUESTED** [review]
Hi Rishabdev,

Thank you for your contribution, your changes are in the right direction but need some adjustments

**Comment by Stefterv** [comment]
Also @SableRaf lmk if the visual changes are approved, especially regarding the spacing and fontsize changes

**Comment by SableRaf** [comment]
Thanks @Stefterv! Yes it looks good to me visually.

File: package.json (hunk 1 above)
Comment (rishabdev2997) [reply package.json:74]: ### When I ran the application it showed this -
**ERROR in polyfill
Module not found: Error: Cannot find module 'react-refresh'**

Comment (rishabdev2997) [reply package.json:74]: Can you suggest what can be done ?

File: src/components/Sidebar.module.css (hunk 2 above)
Comment (rishabdev2997) [reply src/components/Sidebar.module.css:115]: fixed

File: src/components/Sidebar.module.css (hunk 3 above)
Comment (rishabdev2997) [reply src/components/Sidebar.module.css:131]: removed the unused code

File: package.json (hunk 1 above)
Comment (Stefterv) [reply package.json:74]: I can't replicate that issue, what's your browser/system/node version etc...?

File: src/components/Sidebar.module.css [hunk 4]
```diff
@@ -110,20 +110,40 @@
...
+      content: '';
+      position: absolute;
+      left: 0;
+      top: 50%;
+      height: calc(100% - 20%);
+      width: 4px;
+      background-color: transparent;
+      transition: background-color 0.3s;
+      transform: translateY(-50%);
```
Comment (Stefterv) [inline src/components/Sidebar.module.css:140]: This entire section is not used

File: package.json (hunk 1 above)
Comment (rishabdev2997) [reply package.json:74]: browser - chrome
system - windows 11
node - v19.6.1

**Comment by SableRaf** [comment]
Hey @rishabdev2997! Just checking in—are you still interested in finishing up this PR? Let me know; otherwise, I'll go ahead and close it. Thanks!

**Comment by rishabdev2997** [comment]
@SableRaf I will finish it in 3 days. sorry for late reply

**Comment by SableRaf** [comment]
Sure thing! No worries. Thanks for your reply :) let us know if you need any support.

**Comment by rishabdev2997** [comment]
I will make a new pr with the updates
@SableRaf
